src/uiSelectDirective.js

```diff
@@ -274,9 +289,14 @@ uis.directive('uiSelect',
 
               // Determine if the direction of the dropdown needs to be changed.
               if (offset.top + offset.height + offsetDropdown.height > $document[0].documentElement.scrollTop + $document[0].documentElement.clientHeight) {
+                element.addClass(directionUpClassName);
+                offsetDropdown = uisOffset(dropdown); // recalculate after class changed
                 dropdown[0].style.position = 'absolute';
                 dropdown[0].style.top = (offsetDropdown.height * -1) + 'px';
-                element.addClass(directionUpClassName);
+                // Reposition upward dropdown each time the collection of items changes
+                scope.$watchCollection('$select.items', function() {
+                  $timeout(repositionUpwardDropdown);
```

Comment:
There is a big optimization that can be done here. Instead of doing a $timeout everytime it changes, scheduling an operation with `scope.$$postDigest` would batch it. This requires the addition of a boolean check inside the $watchCollection callback, and resetting the flag when the function is executed.

Something like
```js
var foo = false;
scope.$watchCollection('$select.items', function() {
  if (!foo) {
    scope.$$postDigest(function() {
      foo = false;
      repositionUpwardDropdown();
    });
  }
});
```